### PR TITLE
Enable copyright-check and update copyrights

### PR DESCRIPTION
**Pull request by jbescos**
Currently the build doesn't fail when copyright fails because we need to execute this one: glassfish-copyright:check

There are many dates to update.
- Jersey-bot updates pom files version, but the date was not updated. When jersey-bot runs it should invoke 'glassfish-copyright:repair'
- Many files were updated in this commit 'Move CDI integration tests to a common CDI-Integration module (#4280)' but the date was not updated.

### CODE REVIEW

**Comment by senivam** [comment]
actually I do not agree with automatic build failure (that is why it was copyright:copyright). Typical use case is when new Jersey release is published and all pom.xml files are modified and they are being modified with jersey-bot which belongs to Eclipse and has nothing to do with Oracle. Any other contribution from community also does not necessary shall update Oracle's copyright. Payara for instance has its own copyright.
 
Basically the whole community shall not bother with Oracle's copyright, except people from Oracle :)

**Comment by jansupol** [comment]
This is NOT Oracle copyright. This is Oracle and affiliates' copyright. If I understand it correctly, any contributor is understood as an affiliate.

Regarding the release, I believe it is possible to update the copyright year in poms by a sed swarm, is it not?

File: connectors/jetty-connector/src/test/java/org/glassfish/jersey/jetty/connector/ProxyTest.java [hunk 1]
```diff
@@ -1,4 +1,5 @@
 /*
+ * Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
```
Comment (jansupol) [inline connectors/jetty-connector/src/test/java/org/glassfish/jersey/jetty/connector/ProxyTest.java:2]: There seems to be something wrong. This file was not modified in 2020 and yet the copyright plugin requires a copyright change.

Comment (jbescos) [reply connectors/jetty-connector/src/test/java/org/glassfish/jersey/jetty/connector/ProxyTest.java:2]: It is because the plugin removed the section, and this is considered as a new change.
 * Contributors:
 *    Marcelo Rubim

File: examples/entity-filtering-selectable/src/main/java/org/glassfish/jersey/examples/entityfiltering/selectable/App.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2014, 2019Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
```
Comment (jbescos) [inline examples/entity-filtering-selectable/src/main/java/org/glassfish/jersey/examples/entityfiltering/selectable/App.java:2]: This seems wrong, I will fix it.

**Comment by senivam** [comment]
> This is NOT Oracle copyright. This is Oracle and affiliates' copyright. If I understand it correctly, any contributor is understood as an affiliate.

As I remember, this project was contributed to Eclipse, since then any contributor is understood as Eclipse's affiliate. That is why ECA sign is required for any IP contribution.

If it would be Helidon project, then yes, it belongs to Oracle, but this project belongs to Eclipse.

**Comment by m0mus** [comment]
Just want to confirm the copyright should be in this format:
```
Copyright (c) 2002, 2020 Oracle and/or its affiliates. All rights reserved.
```
where `2002` is the year when this file was originally created and `2020` the year of last change.
If the file is new it's just
```
Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
```

**Comment by senivam** [comment]
OK, after all discussions I have to agree with build failure in Travis build due to copyright year changes.

**Review by senivam — APPROVED** [review]

**Comment by jbescos** [comment]
For some reason in Travis it fails in the first module, but there is no information about why it fails. Locally it is working.

I need to push some commits to print more information, but the problem is that Travis seems to be running nightly. It is going to take some time to find why is it failing in Travis.

File: media/json-jackson/src/main/resources/META-INF/NOTICE.markdown [hunk 3]
```diff
@@ -1,4 +1,20 @@
-# Notice for Jersey Json Jackson module 
+#
+# Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
```
Comment (senivam) [inline media/json-jackson/src/main/resources/META-INF/NOTICE.markdown:2]: this is kinda .md file (only with .markdown extension), shall be excluded from Copyright check.

File: pom.xml [hunk 4]
```diff
@@ -642,17 +642,16 @@
...
                     <groupId>org.glassfish.copyright</groupId>
                     <artifactId>glassfish-copyright-maven-plugin</artifactId>
-                    <version>2.0</version>
                     <configuration>
                         <excludeFile>etc/config/copyright-exclude</excludeFile>
                         <!--svn|mercurial|git - defaults to svn-->
                         <scm>git</scm>
                         <!-- turn on/off debugging -->
-                        <debug>false</debug>
+                        <debug>true</debug>
```
Comment (senivam) [inline pom.xml:650]: why in debug mode?

**Review by senivam — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 4 above)
Comment (jbescos) [reply pom.xml:650]: I have added it temporarily to know why is it failing in Travis. Currently it fails but there is no information of the error.

File: media/json-jackson/src/main/resources/META-INF/NOTICE.markdown (hunk 3 above)
Comment (jbescos) [reply media/json-jackson/src/main/resources/META-INF/NOTICE.markdown]: Done, it is also added in the .markdown in the jersey/etc/config/copyright-exclude

File: pom.xml (hunk 4 above)
Comment (senivam) [reply pom.xml:650]: for now build did not even started since the last change :) Travis seems to be really slow recently.
Regarding additional info - it might not show it because of reduced output in Travis. We had to reduce it due to Travis' limitations, otherwise all build threads would fail. May be you can temporary try to run it with full log output (just for copyright-check thread) and this will show what actually happened there?

Comment (senivam) [reply pom.xml:650]: ah, no, copyright already runs with full output, so may be -e key will help when build finally starts :)

Comment (jbescos) [reply pom.xml:650]: I tried it yesterday with -e already but the stacktrace doesn't help, you can see it here:
https://travis-ci.org/github/eclipse-ee4j/jersey/jobs/747370995

Hopefully debug=true will help.

Locally there are no problems. I don't really understand why in Travis it works different.

Comment (senivam) [reply pom.xml:650]: the issue with no output on failures seems to be caused by this hard-coded logic inside the plugin:
https://github.com/eclipse-ee4j/glassfish-copyright-plugin/blob/94c7ba4c6d3151b57f92cf030f77661f1ff15d17/src/main/java/org/glassfish/copyright/CheckCopyrightMojo.java#L33
It's being always quiet in other words. And when you remove this hard-coded line (default is false), you can see why build fails. However fix requires new PR and then new plugin version shall be released...

**Review by senivam — COMMENTED** [review]
current content looks good to me, however due to discovered issue with glassfish-copyright-plugin it's required to fix (or provide some workaround) for error output, and this will introduce some changes in the current PR (like plugin version update or something), so, it shall be merged only after reliable solution for error output inside glassfish-copyright-plugin is provided (and this PR is adjusted accordingly).

**Comment by senivam** [comment]
@jbescos could you please update plugin's version to 2.4?

**Comment by jbescos** [comment]
@senivam finally it is working in Travis

**Review by senivam — APPROVED** [review]
it looks like all passes OK, settings for copyright plugin and related build configuration are adjusted correctly, changes can be merged.

**Comment by jbescos** [comment]
There are some issues with the rebase of my commits, because my commits belongs to 2020 and new files have been modified in 2021. The plugin is not dealing well with that.
